### ISSUE TITLE
Trying to add a value with an unsupported type: array

### ISSUE THREAD

**Issue by srichter** [rap2hpoutre/fast-excel#11]
When I try to export my User model I receive this error:

`Trying to add a value with an unsupported type: array (Box\\Spout\\Common\\Exception\\InvalidArgumentException(code: 0): Trying to add a value with an unsupported type: array at /home/vagrant/dev/vendor/box/spout/src/Spout/Writer/XLSX/Internal/Worksheet.php:231)
`

I believe this is occurring because I eager-load the user's roles which are a collection themselves. If I remove the eager-load it exports fine.

Here is what my fully loaded User model looks like:

```
User::whereId(1)->get();
=> Illuminate\Database\Eloquent\Collection {#903
     all: [
       App\User {#915
         id: 1,
         email: "dummy@data.com",
         name: "Dummy User",
         address: "123 Dummy St",
         city: "City",
         state: "NY",
         zip: "12345",
         phone: "5555555555",
         created_at: "2018-04-01 02:44:42",
         updated_at: "2018-04-02 11:50:38",
         roles: Illuminate\Database\Eloquent\Collection {#884
           all: [
             App\Role {#901
               id: 3,
               name: "admin",
               description: "Administrator",
               created_at: "2018-04-01 02:44:26",
               updated_at: "2018-04-01 02:44:26",
               pivot: Illuminate\Database\Eloquent\Relations\Pivot {#885
                 user_id: 1,
                 role_id: 3,
               },
             },
           ],
         },
       },
     ],
   }


```

**Comment by srichter** [comment]
Sounds good, thanks for the awesome package! For now I will just transform the collection before export. If I can think of a solution to handle nested collections properly I will open a PR

**Comment by rap2hpoutre** [comment]
I fixed it in v0.0.7! You could either use your model with roles and there will be ignored (because this is not string), or map properties:

```php
(new FastExcel(User::all()))->export('users.csv', function ($user) {
    return [
        'Email' => $user->email,
        'Name' => $user->name,

        // Do what you want with roles
        'Last Name' => $user->roles->first()->name
    ];
});
```

Let me know if I can close this issue! Thanks again for your contribution!

**Comment by madsem** [comment]
I have the same issue, but when building sheets with queries that return a LazyCollection:

error:
```log
InvalidArgumentException

  Unsupported type for $data

  at vendor/rap2hpoutre/fast-excel/src/Exportable.php:112
    108▕                 $this->writeRowsFromGenerator($writer, $collection, $callback);
    109▕             } elseif (is_array($collection)) {
    110▕                 $this->writeRowsFromArray($writer, $collection, $callback);
    111▕             } else {
  ➜ 112▕                 throw new InvalidArgumentException('Unsupported type for $data');
    113▕             }
    114▕             if (is_string($key)) {
    115▕                 $writer->getCurrentSheet()->setName($key);
    116▕             }

      +1 vendor frames
  2   app/Jobs/BuildCompanyAdvertisingReportExports.php:82
      Rap2hpoutre\FastExcel\FastExcel::export()
```

And this is my code:
```php
$file = storage_path($this->fileName());
        $header_style = $this->getHeaderStyle();
        $rows_style = $this->getRowStyle();
        $sheets = $this->buildSheets();

        (new FastExcel($sheets))
            ->headerStyle($header_style)
            ->rowsStyle($rows_style)
            ->export($file);

        // Upload file to temporary s3 disk
        $s3->put($this->fileName(), $file, 'tmp');

        $this->company->file_attachments = [$this->fileName()];
        $this->company->last_report_export = now();
        $this->company->save();
```

```php
/**
     * @return \Rap2hpoutre\FastExcel\SheetCollection
     */
    private function buildSheets(): SheetCollection
    {
        return new SheetCollection([
            'Campaign Performance (month)' => CampaignPerformanceReport::byMonth(),
            'Campaign Performance (day)' => CampaignPerformanceReport::byDay(),
        ]);
    }
```

byMonth method:
```php
public static function byMonth(): LazyCollection
    {
        return CampaignPerformanceReport::select([
                'day',
                'remote_campaign_name',
            ])
            ->selectRaw('sum(impressions) as impressions')
            ->selectRaw('sum(clicks) as clicks')
            ->selectRaw('sum(ctr) as ctr')
            ->selectRaw('sum(conversions) as conversions')
            ->selectRaw('sum(conversion_rate) as conversion_rate')
            ->selectRaw('sum(revenue_in_base_currency) as revenue_in_base_currency')
            ->selectRaw('sum(spend_in_base_currency) as spend_in_base_currency')
            ->selectRaw('avg(average_cpc_in_base_currency) as avg_cpc_in_base_currency')
            ->where('company_id', '1eb8e312-7f05-6dc4-9f46-0242ac170004')
            ->groupBy('remote_campaign_name')
            ->lazy();
    }
```

Am I doing this wrong, or is this unexpected?
